### PR TITLE
[MAINTENANCE] gx v1 mirror with windows

### PR DESCRIPTION
**Pull request by rreinoldsc**
A cherry-pick of windows feature that had to be done manually due to the volume of unrelated conflicts.

smoketest:
```
import great_expectations_v1
import great_expectations_v1.expectations.window
```

According to [instructions](https://greatexpectations.atlassian.net/wiki/spaces/~367998638/pages/1218150570/Expectation+Schema+Build+Process), these tests are not passable. To verify in `mercury` branch.

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations_v1/expectations/expectation.py [hunk 1]
```diff
@@ -256,7 +258,7 @@ def __new__(cls, clsname, bases, attrs):
 class Expectation(pydantic.BaseModel, metaclass=MetaExpectation):
     """Base class for all Expectations.
 
-    For a list of all available expectation types, see the `Expectation Gallery <https://greatexpectations.io/expectations/>`_.
+    For a list of all available expectation types, see the `Expectation Gallery <https://great_expectations_v1.io/expectations/>`_.
```
Comment (anthonyburdi) [inline great_expectations_v1/expectations/expectation.py]:
```suggestion
    For a list of all available expectation types, see the `Expectation Gallery <https://great_expectations.io/expectations/>`_.
```

File: great_expectations_v1/expectations/expectation.py [hunk 2]
```diff
@@ -330,12 +332,30 @@ def schema_extra(schema: Dict[str, Any], model: Type[Expectation]) -> None:
...
 
     expectation_type: ClassVar[str]
+    windows: Optional[List[Window]] = pydantic.Field(default=None, description=WINDOWS_DESCRIPTION)
     examples: ClassVar[List[dict]] = []
 
     _save_callback: Union[Callable[[Expectation], Expectation], None] = pydantic.PrivateAttr(
         default=None
     )
 
+    @override
```
Comment (anthonyburdi) [inline great_expectations_v1/expectations/expectation.py:342]: It looks like from here down the changes are not in the PR to v1 to add the windows (https://github.com/great-expectations/great_expectations/pull/10402/files#diff-c9884e297071dade0dfcd31d7473b7afa9ed018adfc18ad7b5183660e4e7ece7). I think that is OK, just calling it out.

File: tests/expectations/test_expectation.py [hunk 3]
```diff
@@ -298,3 +351,66 @@ def test_expectation_with_duplicate_suite_parameters(self):
             max_value={"$PARAMETER": self.SUITE_PARAMETER_VALUE},
         )
         assert expectation.suite_parameter_options == (self.SUITE_PARAMETER_VALUE,)
+
+
+@pytest.mark.unit
```
Comment (anthonyburdi) [inline tests/expectations/test_expectation.py:356]: It looks like from here down the changes are not in the PR to v1 to add the windows (https://github.com/great-expectations/great_expectations/pull/10402/files#diff-c9884e297071dade0dfcd31d7473b7afa9ed018adfc18ad7b5183660e4e7ece7). I think that is OK, just calling it out.

**Review by anthonyburdi — APPROVED** [review]
LGTM! One minor link change.

**Review by NathanFarmer — APPROVED** [review]
It looks good to me

File: tests/expectations/test_expectation.py (hunk 3 above)
Comment (rreinoldsc) [reply tests/expectations/test_expectation.py:356]: These were added while trying to get simple unit tests working, but the branch will simply not pass. It's ok to add

File: great_expectations_v1/expectations/expectation.py (hunk 2 above)
Comment (rreinoldsc) [reply great_expectations_v1/expectations/expectation.py:342]: These were added while trying to get simple unit tests working, but the branch will simply not pass. It's ok to add

**Pull request merged**